From 618a704005f6c00a884d4e6fcbfd842224be3b5e Mon Sep 17 00:00:00 2001
From: Jinping Wang <jinping.wang@amlogic.com>
Date: Tue, 21 May 2019 18:56:20 +0800
Subject: [PATCH] cec: skip Intent.ACTION_SHUTDOWN when reboot [1/1]

PD# SWPL-8666

Problem:
HdmiControlService will receive Intent.ACTION_SHUTDOWN
when reboot, it may lead to send <Standby>

Solution:
skip Intent.ACTION_SHUTDOWN when reboot

Verify:
test ok using p321

Change-Id: I4f636a05dce0d337beb204e5774e03a55640dc62
Signed-off-by: Jinping Wang <jinping.wang@amlogic.com>
---
 services/core/java/com/android/server/hdmi/HdmiControlService.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/hdmi/HdmiControlService.java b/services/core/java/com/android/server/hdmi/HdmiControlService.java
index a20b427..397eb52 100644
--- a/services/core/java/com/android/server/hdmi/HdmiControlService.java
+++ b/services/core/java/com/android/server/hdmi/HdmiControlService.java
@@ -24,6 +24,7 @@ import static com.android.server.hdmi.Constants.OPTION_MHL_ENABLE;
 import static com.android.server.hdmi.Constants.OPTION_MHL_INPUT_SWITCHING;
 import static com.android.server.hdmi.Constants.OPTION_MHL_POWER_CHARGE;
 import static com.android.server.hdmi.Constants.OPTION_MHL_SERVICE_CONTROL;
+import static com.android.server.power.ShutdownThread.SHUTDOWN_ACTION_PROPERTY;
 
 import android.annotation.Nullable;
 import android.content.BroadcastReceiver;
@@ -144,6 +145,7 @@ public class HdmiControlService extends SystemService {
         @Override
         public void onReceive(Context context, Intent intent) {
             assertRunOnServiceThread();
+            boolean isReboot = SystemProperties.get(SHUTDOWN_ACTION_PROPERTY).contains("1");
             switch (intent.getAction()) {
                 case Intent.ACTION_SCREEN_OFF:
                     if (isPowerOnOrTransient()) {
@@ -162,7 +164,7 @@ public class HdmiControlService extends SystemService {
                     }
                     break;
                 case Intent.ACTION_SHUTDOWN:
-                    if (isPowerOnOrTransient()) {
+                    if (isPowerOnOrTransient() && !isReboot) {
                         onStandby(STANDBY_SHUTDOWN);
                     }
                     break;
-- 
2.10.2

